src/resources/views/crud/fields/icon_picker.blade.php

```diff
@@ -72,33 +72,55 @@
+            <link rel="stylesheet" type="text/css" href="{{ $field['font_icon_file_path'] }}">
         <!-- Bootstrap-Iconpicker -->
         <link rel="stylesheet" href="{{ asset('packages/bootstrap-iconpicker/bootstrap-iconpicker/css/bootstrap-iconpicker.min.css') }}"/>
     @endpush
 
     {{-- FIELD EXTRA JS --}}
     @push('crud_fields_scripts')
         <!-- Bootstrap-Iconpicker -->
-        <script type="text/javascript" src="{{ asset('packages/bootstrap-iconpicker/bootstrap-iconpicker/js/bootstrap-iconpicker.bundle.min.js') }}"></script>
 
+        <script type="text/javascript" src="{{ asset('packages/bootstrap-iconpicker/bootstrap-iconpicker/js/bootstrap-iconpicker.bundle.min.js') }}"></script>
+        @if($field['iconset'] == 'lineawesome')
+            <script type="text/javascript" src="{{ asset('packages/backpack/crud/js/fields/iconpicker/line-icons.js') }}"></script>
+        @endif
         {{-- Bootstrap-Iconpicker - set hidden input value --}}
         <script>
             function bpFieldInitIconPickerElement(element) {
                 var $iconset = element.attr('data-iconset');
                 var $iconButton = element.siblings('button[role=icon-selector]');
                 var $icon = element.attr('value');
+                var $customIconSet = null;
+
+                // if the iconset is lineawesome we create our own iconset for the iconpicker.
+                if($iconset == 'lineawesome') {
+                    // we store it in global window object to avoid fetching the same information again
+                    // in case we have two or more iconpickers with lineawesome
+                    if(typeof window.icon_pickerLineIcons !== "undefined") {
+                        $customIconSet = {
+                            iconClass: '',
+                            iconClassFix: '',
+                            icons: window.icon_pickerLineIcons
+                        }
+                    }else{
+                        alert('to use lineawesome with iconpicker you need to republish crud assets.')
+                    }
+                }
+
+                // in case we have a custom set we use it instead of the default iconset.
+                $iconset = $customIconSet !== null ? $customIconSet : $iconset;
 
                 // we explicit init the iconpicker on the button element.
                 // this way we can init the iconpicker in InlineCreate as in future provide aditional configurations.
-                    $($iconButton).iconpicker({
-                        iconset: $iconset,
-                        icon: $icon
-                    });
-
-                    element.siblings('button[role=icon-selector]').on('change', function(e) {
-                        $(this).siblings('input[type=hidden]').val(e.icon);
-                    });
+                $($iconButton).iconpicker({
+                    iconset: $iconset,
```

Comment:
```suggestion
                    iconset: $customIconSet || $iconset,
```
Does this make it worse to understand?